### DISCUSSION TITLE
Internet connection always needed?

### DISCUSSION THREAD

**Discussion by SadE54** [keygen-sh/community#46] [Q&A] [answered]
Hi, I m interested in the product but I would like to know if an internet connection is always needed each time the software is started? Or is there a mechanism, once activated online the first time to store credentials in the licence file, so a internet connection is not needed anymore? 🤔

**Comment by ezekg** [comment, marked as answer]
This entirely depends on your integration. Since Keygen offers an HTTP API, you're in complete control of when and how your application communicates with our API. You can very easily cache license validations so that after first validation, the application will pull from the cache if an internet connection is not available. [This type of caching](https://github.com/keygen-sh/?q=caching&type=all&language=&sort=) would need to be done on your end, since it's very application-dependent, and it's also not something we can provide via our API.

For more powerful offline licensing, we typically recommend using [License Files](https://keygen.sh/docs/api/cryptography/#cryptographic-lic). These allow you to "checkout" a license or machine for a period of time, and allow you to use the full license object (and some of its relationships) in your application. These can be stored in the client filesystem or environment.

**Reply by SadE54** [reply]
Thanks for the answer.
But I'm a little lost with the dashboard and cannot find a way to create a license file from the license section 😣
Edit: In fact I'm not sure it's possible to get a license file from the dashboard , but only using the API ... if' it' not possible , I guess there's a reason ?

**Reply by ezekg** [reply]
Currently, it's not possible to checkout a license from the dashboard. Because checking out a License File has various options, it will need to be done via the API. We have plans to introduce this into the UI, but no definite timeline on that yet.
